### DISCUSSION TITLE
reload=True teardown

### DISCUSSION THREAD

**Discussion by CiberNin** [Kludex/uvicorn#1614] [Q&A]
Is there any way to ensure code runs before uvicorn reloads?
Something like a finally block?
I have a bunch of physical devices I'm communicating with that each have to be manually restarted if I break the connection without a disconnect message.

**Comment by Kludex** [comment]
Check: https://github.com/encode/uvicorn/pull/1572/

**Reply by CiberNin** [reply]
I read through that and I don't understand how it's supposed to answer my question.
I am running uvicorn programmatically of course. What I want to know is if there is any way to have some code that will run before uvicorn does a reload after it detects that I have saved a file.
